lib/hanami/setup.rb

```diff
@@ -0,0 +1,11 @@
+# frozen_string_literal: true
+
+require "bundler/setup"
+require "hanami"
+
+begin
+  root = Dir.pwd
```

Comment:
As @solnic pointed above, is there the need here as well to make this configurable?

Reply:
@jodosha We can't make this configurable because at this point, we're yet to even have loaded the application class! All we can do at this point is make a _best guess_ as to where that file may reside, based on established conventions (i.e. "try load config/application.rb").

Like I mentioned in my original comment, this could be made more sophisticated by copying Bundler's `require "bundler/setup"` behaviour (where it tries recursing up directories until it finds the file that it wants). However, I think that should definitely be a after-this-PR thing.

I will make _one_ change to this file now, though, where it looks up the location of the routes file from the application config, once it's loaded that.

Reply:
FWIW, the alternative to `require "hanami/setup"`, of course, is something like `require_relative "config/application"` (or whichever relative path is appropriate for the current file), but I think the `require "hanami/setup"` file is a much nicer looking thing, and a much easier to remember thing, if we can get the behaviour right :)